### DISCUSSION TITLE
Feature: double and triple banger custom measurements

### DISCUSSION THREAD

**Discussion by Glutnix** [kazad/instacalc#55] [Ideas]
Just discovered this: 
```
181cm as tape measure     > 5' 11 17⁄64"
```

(might be good to put this example in the unit conversion list as well as in the formatting section where i found it)

but it would be good to be able to do this for other units:

```
10 kg in pounds,stones,ounces   > 1 stone 8 pounds 0.7396 ounces
```

(workaround: you can do this by using rounddown(), but it's clunky and needs 4 lines to get the separate unites)

other examples: 
* yards, feet, inches
* stones, pounds, ounces
* barrels, gallons, pints, floz
* shots, splashes, dashes
It would be great to able to make custom units for more arcane measurements. 

For example, land surveying in European countries historically used Acres, Roods and Perches, which you can still find on old land deeds referred to by surveyors today. (1 acre = 4 roods, 1 rood = 40 perches, 2a3r10.63p = 11,397.7182m^2)

```
rood is (1 acre / 4)
perch is (1 rood / 40)
// note 'perchs' below is wrong, should be 'perches'
2 acres 3 roods 10.63 perchs in m^2    > 11,397.7182m^2 
11,397.7182m^2 in acres, roods, perches > 2 acres, 3 roods, 10.63 perches
```

This also brings up the problem of defining custom units plural and symbol forms.
```
rood,roods,r is (1 acre / 4)
// or a different syntax:
perch(-es, #p) is (1 rood / 40)

2ac3r10.63p in m^2 > 11,397.7182m^2
```

here's the math for ft, in and acres, roods, perches and stones, pounds, ounces, because SEO
https://instacalc.com/56805

**Comment by Glutnix** [comment]
also, i know internationalization is a royal pain, so who knows how to do this right *shrug*

**Comment by kazad** [comment]
Thanks for the detailed feature request!

1. I've updated the help reference text, good catch

2. I've been thinking about formatting for "grouped" units (yard/feet/inches, ton/stone/lbs,oz,). I hadn't thought of custom formats. I might have a general "imperial" format which figures out the right grouping based on distance, weight, etc., and perhaps with the option to give a set of units as a customization.

3. Custom unit definitions are a good feature too. As you noted, you can define variables like "rood = acre / 4" but it doesn't treat "rood" as true unit in conversions (`15000 m^2 to rood`). I'd like to deduce this implicitly, without needing an explicit `unit("rood", acre/4)` -- I like the "don't make me think" philosophy, and if something looks like a unit, it should probably become a unit.

Appreciate the shared calc, that definitely helps with the use case. I'll noodle this over.

**Comment by kazad** [comment]
Quick update, there's now better handling for implicit/custom units: https://instacalc.com/57732

```
rood is (1 acre / 4)      = 0.25 acre
6 acres in rood           = 24 rood
6 acres in roods          = 24 rood
```

* It handles plurals (rood ~ roods)
* You can use the "is" (from before), or just rood = acre / 4

A future task is to figure out the `unit1, unit2, unit3` notation to split up the result into that format.
